File: src/main/java/com/projectswg/forwarder/resources/networking/packets/PingPacket.java
package com.projectswg.forwarder.resources.networking.packets;

import com.projectswg.common.utilities.ByteUtilities;

import java.nio.ByteBuffer;

public class PingPacket extends Packet {
	
	private byte [] payload;
	
	public PingPacket(byte [] payload) {
		this.payload = payload;
	}
	
	@Override
	public void decode(ByteBuffer data) {
		this.payload = data.array();
	}
	
	@Override
	public ByteBuffer encode() {
		return ByteBuffer.wrap(payload);
	}
	
	public byte [] getPayload() {
		return payload;
	}
	
	@Override
	public String toString() {
		return "PingPacket[payload=" + ByteUtilities.getHexString(payload) + "]";
	}
	
}

File: src/main/java/com/projectswg/forwarder/resources/networking/packets/PingPacket.kt
package com.projectswg.forwarder.resources.networking.packets

import com.projectswg.common.network.NetBuffer
import com.projectswg.common.utilities.ByteUtilities

class PingPacket(var payload: ByteArray) : Packet() {
	
	override fun decode(data: NetBuffer) {
		this.payload = data.array()
	}
	
	override fun encode(): NetBuffer {
		return NetBuffer.wrap(payload)
	}
	
	override fun toString(): String {
		return "PingPacket[payload=" + ByteUtilities.getHexString(payload) + "]"
	}
	
}
